### PR TITLE
Changes for simulations over land

### PR DESCRIPTION
**Pull request by sjboeing**

### CODE REVIEW

**Comment by sjboeing** [comment]
Draft pull request with changes to make it possible to run set up land cases. z=0 currently still corresponds to sea level.

**Comment by sjboeing** [comment]
Hi @leifdenby. Here is the Cardington pull request, where I needed to make some further changes. I will continue working on making the examples consistent as a next step.

**Comment by sjboeing** [comment]
@leifdenby: let me know if you have time to take a look at this. It is probably one of the last pieces of functionality we still need, although I think we may still want to have a quick chat about what to do with the height of the surface and e.g. the surface pressure for cases over land.

**Comment by sjboeing** [comment]
One this is merged in, I will come up with a set of suggested case studies.

**Comment by sjboeing** [comment]
Hi @leifdenby . Would you be able to have a look at these changes sometime next week? I will work on updating the case setups and README in parallel to this.

File: lagtraj/utils/gradient_calculation.py [hunk 1]
```diff
@@ -15,35 +15,41 @@ def _boundary_gradients(x_array, y_array, val_array):
     len_lons = np.shape(val_array)[3]
     x_gradient_array = np.empty((len_temp, len_levels))
     y_gradient_array = np.empty((len_temp, len_levels))
-    dval_dx = np.empty((len_lats))
-    dval_dy = np.empty((len_lons))
     for this_time in range(len_temp):
         for this_level in range(len_levels):
-            # This calculates the x-gradient at each latitude
+            dxval_tot = 0.0
```
Comment (leifdenby) [inline lagtraj/utils/gradient_calculation.py:20]: I don't quite understand the need for changes in this file. Is that related to changes in how the mask is done? Or are these improvements for speed (reducing the number of calculations)?

**Review by leifdenby — COMMENTED** [review]
This looks great in general @sjboeing. I think there are just a few things we could talk about to clarify a bit. I think the mask creation and application of the masks could be done a little differently, but shouldn't take long to discuss :)

**Comment by sjboeing** [comment]
Hi @leifdenby. I have addressed the issues you mentioned, and have just added a new issue for removing the "all" mask (which I propose to look into as a separate item, running without a mask has given me trouble in the past). I will wait for checks to pass, just let me know if you are happy for me to merge this.

**Comment by sjboeing** [comment]
Hi @leifdenby: just let me know if this is good to go.

**Comment by leifdenby** [comment]
Looks great @sjboeing, let's get it merged in!

File: lagtraj/domain/mask.py [hunk 2]
```diff
@@ -14,8 +15,29 @@ def _compute_ocean_only_mask(ds, z_max=5.0, lsm_max=0.2):
...
+    ] = "Land only (land-sea mask > {lsm_min} [{ds.lsm.units}])"
+    return da_mask
+
+
 def calc_mask(ds, mask_type):
     if mask_type == "ocean_only":
         return _compute_ocean_only_mask(ds=ds)
+    elif mask_type == "land_only":
+        return _compute_land_only_mask(ds=ds)
+    elif mask_type == "all":
```
[the remark this answers is not in the document]
Comment (leifdenby) [reply lagtraj/domain/mask.py:42]: We'll leave this for now I think, but I'll open a new issue in case we want to clean this up a bit

**Review by leifdenby — APPROVED** [review]
